### ISSUE TITLE
Error thrown accessing property of null relationship (e.g. after an OPTIONAL MATCH)

### ISSUE THREAD

**Issue by plasticfist** [neo4j/neo4j#12573]
- Neo4j version: 4.1.0 and 4.1.1
- Operating system: NA
- API/Driver: NA

- **Steps to reproduce**
Use the dataset from https://neo4j.com/docs/graph-data-science/current/alpha-algorithms/cosine/#algorithms-similarity-cosine-procedure-sample, or any trivial graph, just make sure there are null relationships coming back in the OPTIONAL MATCH

then run this cypher
//demonstrate bug
MATCH (p:Person), (c:Cuisine)
 OPTIONAL MATCH (p)-[likes:LIKES]->(c)
 return likes.score

- Expected behavior (from Neo4j 4.0.7)

likes.score
--
1 | 10
2 | null
3 | 6
4 | null
5 | null
6 | 2
7 | null
8 | null
9 | null
10 | 9
11 | null
12 | 7
13 | 8
14 | 1
15 | 8
16 | 7
17 | 9
18 | null
...

- Actual behavior (verified bug exists in Neo4j 4.1.0 and 4.1.1)

Neo.DatabaseError.Statement.ExecutionFailed

Additional References:
https://community.neo4j.com/t/cosine-similarity-example-unexpected-results/23630/2
https://community.neo4j.com/t/upgrade-from-4-0-to-4-1-causing-errors-when-using-with-with-collect/23984

**Comment by mnd999** [comment]
Thanks for the report, we'll look into it

**Comment by grzegorz-aniol** [comment]
:+1: 
I confirm that issue and I'm voting for solving this 

Here is the exception:
```
Caused by: java.lang.NullPointerException
        at org.neo4j.kernel.impl.newapi.DefaultPropertyCursor.initRelationship(DefaultPropertyCursor.java:97)
        at org.neo4j.kernel.impl.newapi.DefaultRelationshipCursor.properties(DefaultRelationshipCursor.java:83)
        at org.neo4j.internal.kernel.api.helpers.CachingExpandInto$ExpandIntoSelectionCursor.properties(CachingExpandInto.java:538)
        at org.neo4j.codegen.OperatorTask70.compiledOperate(Unknown Source)
        at org.neo4j.cypher.internal.runtime.pipelined.operators.CompiledTask.operateWithProfile(OperatorCodeGenBaseTemplates.scala:247)
        at org.neo4j.cypher.internal.runtime.pipelined.PipelineTask.executeOperators(PipelineTask.scala:62)
        at org.neo4j.cypher.internal.runtime.pipelined.PipelineTask.executeWorkUnit(PipelineTask.scala:52)
        at org.neo4j.cypher.internal.runtime.pipelined.Worker.executeTask(Worker.scala:135)        at org.neo4j.cypher.internal.runtime.pipelined.Worker.workOnQuery(Worker.scala:91)
        at org.neo4j.cypher.internal.runtime.pipelined.execution.CallingThreadExecutingQuery.request(CallingThreadExecutingQuery.scala:41)
        at org.neo4j.cypher.internal.PipelinedRuntime$PipelinedRuntimeResult.request(PipelinedRuntime.scala:426)
        at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:99)
        at org.neo4j.cypher.internal.result.ClosingExecutionResult.request(ClosingExecutionResult.scala:145)
        at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.doRequest(QuerySubject.java:184)
        at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.request(QuerySubject.java:167)
        at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
        at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
        at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
        at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
        at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:130)
        at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.request(Operators.java:2089)
        at reactor.core.publisher.StrictSubscriber.request(StrictSubscriber.java:138)
        at org.neo4j.fabric.stream.Rx2SyncStream$RecordSubscriber.request(Rx2SyncStream.java:129)
        at org.neo4j.fabric.stream.Rx2SyncStream.maybeRequest(Rx2SyncStream.java:91)
        at org.neo4j.fabric.stream.Rx2SyncStream.readRecord(Rx2SyncStream.java:50)
```

And here is the query from the documentation:
```cypher
MATCH (p:Person), (c:Cuisine)
 OPTIONAL MATCH (p)-[likes:LIKES]->(c)
 WITH {item:id(p), weights: collect(coalesce(likes.score, gds.util.NaN()))} AS userData
 WITH collect(userData) AS d1
 CALL gds.alpha.similarity.cosine.stream({data: d1})
 YIELD item1, item2, count1, count2, similarity
 RETURN gds.util.asNode(item1).name AS from, gds.util.asNode(item2).name AS to, similarity
 ORDER BY similarity DESC
```

**Comment by mnd999** [comment]
This issue should be resolved in Neo4j 4.1.2. The release notes are here https://github.com/neo4j/neo4j/wiki/Neo4j-4.1-changelog#cypher. If this does not resolve the issue or you are still having problems feel free to reopen.
